Assets/MixedRealityToolkit/Providers/Hands/AltHandRay.cs

```diff
@@ -0,0 +1,151 @@
+﻿using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// Alternate hand ray implementation by prvncher (Eric Provencher)
+    ///
+    /// What does this do?
+    /// - Rather than computing hand ray direction by estimating shoulder position by offsetting a point using the head rotation,
+    /// We estimate a torso direction, using head and hand positions, and use that to estimate the shoulder offset.
+    /// 
+    /// Why use this class instead of the original hand ray?
+    /// - More resilient to head rotating while working with menu
+    /// - Overall more stable, with a more comfortable angle
+    ///
+    /// How to use this class?
+    /// - Right now, there is no mechanism in place to introduce an alternate hand ray implementation.
+    /// - For now, to get this working, simply replace the class HandRay, by AltHandRay in BaseHand.cs.
+    /// </summary>
+    public class AltHandRay
```

Comment:
It may be worth defining an interface at this point, which we could use for both this class and the existing `HandRay` class. Could also be done when we do the work to support arbitrary hand ray types without editing scripts.

Reply:
Added IHandRay

Reply:
Awesome, thanks!